src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -139,22 +139,6 @@ public static void DictionaryOfString()
             }
         }
 
-        [Fact]
-        public static void DuplicateKeysFail()
-        {
-            // Non-generic IDictionary case.
-            Assert.Throws<JsonException>(() => JsonSerializer.Parse<IDictionary>(
```

Comment:
Can we add this test case back? We lost the `IDictionary` test as part of this change (i.e. non-generic IDictionary), and I am not sure why.

Also, we should add tests for the generic `IDictionary<TKey, TValue>` as well (rather than just a concrete `Dictionary<TKey, TValue>` type).